File: frontend/test_favorite.js
const axios = require('axios');

// 测试收藏功能的脚本
async function testFavorite() {
  try {
    // 使用之前测试成功的token
    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiMTAwMDEiLCJ1c2VybmFtZSI6InRlc3R1c2VyX25ldzQ1NiIsImV4cCI6MTcxNDg2NjEyM30.D6eZ2Cj7I9B9eRzXZ1DcV2FgH5E7J8K9L0M1N2O3P4Q5R6S7T8U9V0W1X2Y3Z4';
    
    // 创建axios实例
    const api = axios.create({
      baseURL: 'http://localhost:8001',
      headers: {
        Authorization: `Bearer ${token}`
      }
    });
    
    console.log('测试添加收藏...');
    const addResponse = await api.post('/recipes/1/favorite');
    console.log('添加收藏成功:', addResponse.status);
    
    console.log('测试取消收藏...');
    const removeResponse = await api.delete('/recipes/1/favorite');
    console.log('取消收藏成功:', removeResponse.status);
    
    console.log('测试完成！收藏功能正常工作。');
  } catch (error) {
    console.error('测试失败:', error.response?.status, error.response?.data);
  }
}

testFavorite();

File: test_favorite.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import requests
import json

# 登录请求
def login():
    url = "http://localhost:8002/api/auth/login"
    headers = {"Content-Type": "application/json"}
    data = {"username": "testuser2", "password": "Test123!"}
    
    try:
        response = requests.post(url, headers=headers, json=data)
        response.raise_for_status()  # 如果响应状态码不是200，抛出异常
        
        print(f"登录成功，状态码: {response.status_code}")
        
        # 尝试解析响应内容
        try:
            json_response = response.json()
            print("响应内容:")
            print(json.dumps(json_response, indent=2, ensure_ascii=False))
            
            # 检查是否有访问令牌
            if "data" in json_response and "access_token" in json_response["data"]:
                print(f"\n访问令牌: {json_response['data']['access_token']}")
                return json_response['data']['access_token']
            else:
                print("\n响应中没有访问令牌")
                return None
                
        except json.JSONDecodeError as e:
            print(f"解析JSON响应失败: {e}")
            print(f"原始响应内容: {response.text}")
            return None
            
    except requests.exceptions.RequestException as e:
        print(f"登录请求失败: {e}")
        return None

# 测试收藏功能
def test_favorite(access_token, recipe_id):
    if not access_token:
        print("没有访问令牌，无法测试收藏功能")
        return
    
    url = f"http://localhost:8002/api/recipes/{recipe_id}/favorite"
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {access_token}"
    }
    
    try:
        response = requests.post(url, headers=headers, json={})
        
        print(f"\n收藏食谱请求，状态码: {response.status_code}")
        
        try:
            json_response = response.json()
            print("响应内容:")
            print(json.dumps(json_response, indent=2, ensure_ascii=False))
        except json.JSONDecodeError:
            print(f"原始响应内容: {response.text}")
            
    except requests.exceptions.RequestException as e:
        print(f"收藏请求失败: {e}")

if __name__ == "__main__":
    # 测试登录
    token = login()
    
    # 测试收藏功能（使用一个示例食谱ID）
    if token:
        test_recipe_id = "583d7d70-b89c-494b-8d66-0e60f8484428"
        test_favorite(token, test_recipe_id)
